md-dao/src/main/java/com/wipro/ats/bdre/md/dao/GeneralConfigDAO.java

```diff
@@ -179,6 +181,67 @@ public void delete(GeneralConfigId generalConfigId) {
         return generalConfigList;
     }
 
+    public List<com.wipro.ats.bdre.md.beans.table.GeneralConfig> getLikeGeneralConfig(String description, int required) {
+        Session session = sessionFactory.openSession();
+        List<com.wipro.ats.bdre.md.beans.table.GeneralConfig> generalConfigList = new ArrayList<com.wipro.ats.bdre.md.beans.table.GeneralConfig>();
+
+        try {
+            session.beginTransaction();
+
+            if (required == 2) {
+                Criteria getGeneralConfigCriteria = session.createCriteria(GeneralConfig.class).add(Restrictions.like(CONFIG_GROUP, "cluster%")).add(Restrictions.eq("description", description)).add(Restrictions.eq("enabled", true));
+                List<GeneralConfig> jpaGeneralConfigList = getGeneralConfigCriteria.list();
+                for (GeneralConfig jpaGeneralConfig : jpaGeneralConfigList) {
+                    com.wipro.ats.bdre.md.beans.table.GeneralConfig generalConfig = new com.wipro.ats.bdre.md.beans.table.GeneralConfig();
+                    generalConfig.setCounter(jpaGeneralConfigList.size());
+                    if (jpaGeneralConfig.getRequired())
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS134)